### PR TITLE
Suggestions from review SHM homework 1

### PR DESCRIPTION
**Pull request by arafalski**

### CODE REVIEW

**Review by kontrybutor — APPROVED** [review]

File: shm/inc/map.hpp [hunk 1]
```diff
@@ -10,7 +10,6 @@ class Map {
     Island* getIsland(const Coordinates& coordinates);
 
 private:
-    void generateIslands();
```
Comment (kontrybutor) [inline shm/inc/map.hpp]: Why did you remove it?

File: shm/src/map.cpp [hunk 2]
```diff
@@ -10,7 +10,13 @@ constexpr size_t kMapWidth = 50;
...
 
 Map::Map() {
-    generateIslands();
+    std::random_device rd;
+    std::mt19937 eng(rd());
+    std::uniform_int_distribution<> distribX(0, kMapWidth - 1);
+    std::uniform_int_distribution<> distribY(0, kMapHeight - 1);
+
+    islands_.resize(kIslandAmt);
+    std::generate(islands_.begin(), islands_.end(), [&]() { return Island(distribX(eng), distribY(eng)); });
```
Comment (kontrybutor) [inline shm/src/map.cpp]: Are you sure that this won't generate Islands with the same coordinates?

Comment (arafalski) [reply shm/src/map.cpp]: I didn't think about it. Ok I have another idea. That I will try out :)

File: shm/inc/map.hpp (hunk 1 above)
Comment (arafalski) [reply shm/inc/map.hpp]: Never mind

File: shm/src/map.cpp (hunk 2 above)
Comment (kontrybutor) [reply shm/src/map.cpp]: I've proposed solution with uniform_int_distribution but our team chose vector approach :)

Comment (kontrybutor) [reply shm/src/map.cpp]:
```suggestion
    std::random_device rd;
    std::mt19937 eng(rd());
    islands_.reserve(kIslandAmt);

    std::uniform_int_distribution<> widthGen(0, kMapWidth);
    std::uniform_int_distribution<> heightGen(0, kMapHeight);

    std::set<std::pair<int, int>> uniqueIslands;
    int x = 0;
    int y = 0;
    while (unique_islands.size() < kIslandAmt) {
        x = widthGen(eng);
        y = heightGen(eng);

        uniqueIslands.insert({x, y});
    }
    std::transform(uniqueIslands.begin(), uniqueIslands.end(),
                   std::back_inserter(islands_),
                   [](const auto& coords) {
                       return Island(coords.first, coords.second);
                   });
```
That was my proposition ;)

Comment (kontrybutor) [reply shm/src/map.cpp]: Anyway I am not sure if we have to follow every suggestion from PR in coders-school :)

**Review by kontrybutor — APPROVED** [review]
